tests/load_balancer_test.go

```diff
@@ -0,0 +1,264 @@
+//
+// DISCLAIMER
+//
+// Copyright 2018 ArangoDB GmbH, Cologne, Germany
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Copyright holder is ArangoDB GmbH, Cologne, Germany
+//
+// Author Ewout Prangsma
+//
+
+package tests
+
+import (
+	"context"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/dchest/uniuri"
+	"github.com/stretchr/testify/assert"
+
+	driver "github.com/arangodb/go-driver"
+	api "github.com/arangodb/kube-arangodb/pkg/apis/deployment/v1alpha"
+	"github.com/arangodb/kube-arangodb/pkg/client"
+	"github.com/arangodb/kube-arangodb/pkg/util"
+)
+
+type queryTest struct {
+	Query             string
+	BindVars          map[string]interface{}
+	ExpectSuccess     bool
+	ExpectedDocuments []interface{}
+	DocumentType      reflect.Type
+}
+
+type queryTestContext struct {
+	Context     context.Context
+	ExpectCount bool
+}
+
+
+// TestLoadBalancerCursorVST tests cursor forwarding with load-balanced VST
+func TestLoadBalancerCursorVST(t *testing.T) {
```

Comment:
1. Set the use of VST explicitly, not via context.
2. I would "duplicate" this test for VST & HTTP. To do so, create a helper function that contains almost all of the test and from `TestLoadBalancerCursorVST` and `TestLoadBalancerCursorHTTP` call that function with 1 argument specifying VST/HTTP.